File: UI/espacenter_mobile/lib/models/termin.dart
import 'package:espacenter_mobile/models/rezervacija.dart';
import 'package:espacenter_mobile/models/usluga.dart';
import 'package:flutter/material.dart';
import 'package:json_annotation/json_annotation.dart';

import 'korisnik.dart';

part 'termin.g.dart';

@JsonSerializable()
class Termin {
  int? terminID;
  int? korisnikID;
  DateTime? datumTermina;
  String? vrijemeTermina;
  Korisnik? korisnik;
  bool? isBooked;
  bool? isDeleted;
  Rezervacija? rezervacija;
  Usluga? selectedUsluga;

  Termin(this.terminID, this.korisnikID, this.datumTermina, this.vrijemeTermina,
      this.korisnik, this.isDeleted, this.isBooked,this.rezervacija,this.selectedUsluga);

  /// A necessary factory constructor for creating a new User instance
  /// from a map. Pass the map to the generated `_$UserFromJson()` constructor.
  /// The constructor is named after the source class, in this case, User.
  factory Termin.fromJson(Map<String, dynamic> json) => _$TerminFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$TerminToJson(this);
}

File: eSpaCenter.Services/Database/Termin.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eSpaCenter.Services.Database
{
    public class Termin
    {
        public int TerminID { get; set; }
        public DateTime DatumTermina { get; set; }
        public DateTime DatumKreiranja { get; set; } = DateTime.Now.Date;
        public string VrijemeTermina { get; set; } = null!;
        public bool IsBooked { get; set; }
        public int KorisnikID { get; set; }
        public bool isDeleted { get; set; }


        public virtual Korisnik Korisnik { get; set; } = null!;
    }
}
 